### PR TITLE
py-keyring: add new package

### PR DESCRIPTION
**Pull request by adamjstewart**
Successfully builds on macOS 10.15.7 with Python 3.8.8 and Apple Clang 12.0.0.

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-keyring/package.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+
+class PyKeyring(PythonPackage):
+    """The Python keyring library provides an easy way to access the system keyring
+    service from python. It can be used in any application that needs safe password
+    storage."""
+
+    homepage = "https://github.com/jaraco/keyring"
+    pypi     = "keyring/keyring-23.0.1.tar.gz"
+
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/py-keyring/package.py:16]: Are you willing to be the package maintainer?

Comment (tldahlgren) [reply var/spack/repos/builtin/packages/py-keyring/package.py:16]: Nvm.  I vaguely recall I've asked you this before. :)

**Review by tldahlgren — APPROVED** [review]
LGTM

File: var/spack/repos/builtin/packages/py-keyring/package.py (hunk 1 above)
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-keyring/package.py:16]: I'm technically the maintainer for all Python packages

**Pull request merged**